ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -257,6 +553,92 @@ const triggerSet: TriggerSet<Data> = {
+      netRegexKo: NetRegexes.ability({ id: '63E1', source: '기사신 토르당', capture: false }),
+      condition: (data) => data.phase === 'thordan',
+      delaySeconds: 4.7,
+      promise: async (data) => {
+        // The two gladiators spawn in one of two positions: West (95, 100) or East (105, 100).
+        // This triggers uses east/west location of the white knight, Ser Janlennoux (3635) to
+        // determine where both knights will dash as they each only look in two directions.
+        // Ser Janlenoux only looks towards the north (northeast or northwest).
+        // Ser Adelphel only looks towards the south (southwest or southeast).
+        // Thus we can just use one knight and get whether it is east or west for location.
+        // Callout assumes starting from DRK position, but for east/west (two-movement strategy)
+        // you can reverse the cw/ccw callout.
+        const janlenouxLocaleNames: LocaleText = {
+          en: 'Ser Janlenoux',
+        };
+
+        // Select Ser Janlenoux
+        let combatantNameJanlenoux = null;
+        combatantNameJanlenoux = janlenouxLocaleNames[data.parserLang];
+
+        let combatantDataJanlenoux = null;
+        if (combatantNameJanlenoux) {
+          combatantDataJanlenoux = await callOverlayHandler({
+            call: 'getCombatants',
+            names: [combatantNameJanlenoux],
+          });
+        }
+
+        // if we could not retrieve combatant data, the
+        // trigger will not work, so just resume promise here
+        if (combatantDataJanlenoux === null) {
+          console.error(`Ser Janlenoux: null data`);
+          return;
+        }
+        if (!combatantDataJanlenoux.combatants) {
+          console.error(`Ser Janlenoux: null combatants`);
+          return;
+        }
+        const combatantDataJanlenouxLength = combatantDataJanlenoux.combatants.length;
+        if (combatantDataJanlenouxLength <= 1) {
+          console.error(`Ser Janlenoux: expected at least 1 combatants got ${combatantDataJanlenouxLength}`);
+          return;
+        }
+
+        // Sort to retreive last combatant in list
+        const sortCombatants = (a: PluginCombatantState, b: PluginCombatantState) => (a.ID ?? 0) - (b.ID ?? 0);
+        const combatantJanlenoux = combatantDataJanlenoux.combatants.sort(sortCombatants).shift();
+        if (!combatantJanlenoux)
+          throw new UnreachableCode();
+
+        data.sanctityWardDir = matchedPositionToDir(combatantJanlenoux);
+      },
+      infoText: (data, _matches, output) => {
+        // Map of directions
+        const dirs: { [dir: number]: string } = {
+          0: output.unknown!(),
+          1: output.unknown!(), // north position
+          2: output.unknown!(),
+          3: output.counterclock!(),
+          4: output.unknown!(),
+          5: output.unknown!(), // south position
+          6: output.unknown!(),
+          7: output.clockwise!(),
+          8: output.unknown!(),
+        };
```

Comment:
I think if east and west are the only valid positions, then I think it's better to just compare x position against the center rather than get a direction (6 of which are invalid).

I am not worried about multiple strategies here, as in this case if it's backwards then folks can change the text in the cactbot config ui easily.  I think clockwise is still better than left though, because folks are always confused about "is it left about looking in or looking out" etc etc.